### PR TITLE
Commendation Letter for FA program

### LINKED ISSUE #4421: enable commendation letter for FA courses
The DEDP course team would like to offer commendation letters to learners, similar to what we have implemented for SCM.

### PR DESCRIPTION
**Pull request by annagav**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Fix  #4421

#### What's this PR do?
Enable commendation letters for learners in FA programs

#### How should this be manually tested?
Enroll your learner in both FA and non-FA program, get all courses passed (with certificates for FA program). Check the dashboard to see if the progress widget has a link to the letter.
If it an FA program the learner must have a program certificate.

Screen Shot 2019-11-14 at 3 18 51 PM

### CODE REVIEW

File: grades/signals.py [hunk 1]
```diff
@@ -20,6 +20,16 @@ def handle_create_coursecertificate(sender, instance, created, **kwargs):  # pyl
...
 
 
+@receiver(post_save, sender=MicromastersProgramCertificate, dispatch_uid="programcertificate_post_save")
+def handle_create_programcertificate(sender, instance, created, **kwargs):  # pylint: disable=unused-argument
+    """
+    When a MicromastersProgramCertificate model is created
+    """
+    if created:
+        user = instance.user
+        transaction.on_commit(lambda: generate_program_letter(user, instance.course.program))
```
Comment (tasawernawaz) [inline grades/signals.py]: I tried to enroll a user in all courses of a program. Then I tried to create a certificate for each course. Got an error in the signals on the last certificate creation.
```
AttributeError: 'MicromastersProgramCertificate' object has no attribute 'course'
```
Let me know if I missed anything here.

**Comment by annagav** [comment]
@tasawernawaz I addressed your comments. Thanks!

**Comment by tasawernawaz** [comment]
@annagav I am still unable to view a program letter for my program. I am trying to create a certificate from the shell. Could you please tell me how you tried this?

Also could you please update the management command at https://github.com/mitodl/micromasters/blob/master/grades/management/commands/generate_program_letter.py

**Comment by annagav** [comment]
@tasawernawaz was the letter not created or it was created but you cannot see it in the dashboard?

**Comment by tasawernawaz** [comment]
@annagav I don't have right now, as I already have a program certificate, could you please update the management command mentioned in https://github.com/mitodl/micromasters/pull/4458#issuecomment-555540873. That command would create later for users who have already earned a program certificate.

**Comment by annagav** [comment]
@tasawernawaz, Thanks for the feedback! I addressed your comments.

**Pull request merged**